server/bin/pbench-index.py

```diff
@@ -102,130 +89,39 @@ def main(options, name):
          8 - Unable to load and process expected mapping files
          9 - Unable to update index templates in configured Elasticsearch
              instance
-    """
-    _name_suf = "-tool-data" if options.index_tool_data else ""
-    _name_re = "-re" if options.re_index else ""
-    name = f"{name}{_name_re}{_name_suf}"
-
-    if not options.cfg_name:
-        print(
-            f"{name}: ERROR: No config file specified; set"
-            " _PBENCH_SERVER_CONFIG env variable or"
-            " use --config <file> on the command line",
-            file=sys.stderr,
-        )
-        return 2
-
-    idxctx = None
-    try:
-        idxctx = IdxContext(options, name, _dbg=_DEBUG)
-    except (ConfigFileError, ConfigParserError) as e:
-        print(f"{name}: {e}", file=sys.stderr)
-        return 2
-    except BadConfig as e:
-        print(f"{name}: {e}", file=sys.stderr)
-        return 3
-    except JsonFileError as e:
-        print(f"{name}: {e}", file=sys.stderr)
-        return 8
-
-    if options.dump_index_patterns:
-        idxctx.templates.dump_idx_patterns()
-        return 0
-
-    if options.dump_templates:
-        idxctx.templates.dump_templates()
-        return 0
-
-    _re_idx = "RE-" if options.re_index else ""
-    if options.index_tool_data:
-        # The link source and destination for the operation of this script
-        # when it only indexes tool data.
-        linksrc = "TO-INDEX-TOOL"
-        linkdest = "INDEXED"
-    else:
-        # The link source and destination for the operation of this script
-        # when it indexes run, table-of-contents, and result data.
-        linksrc = f"TO-{_re_idx}INDEX"
-        linkdest = "TO-INDEX-TOOL"
-    # We only ever use a symlink'd error destination for indexing
-    # problems.
-    linkerrdest = "WONT-INDEX"
-
-    res = 0
+"""
 
-    ARCHIVE_rp = idxctx.config.ARCHIVE
 
-    INCOMING_rp = idxctx.config.INCOMING
-    INCOMING_path = idxctx.config.get_valid_dir_option(
-        "INCOMING", INCOMING_rp, idxctx.logger
-    )
-    if not INCOMING_path:
-        res = 3
-
-    qdir = idxctx.config.get_conf(
-        "QUARANTINE", "pbench-server", "pbench-quarantine-dir", idxctx.logger
-    )
-    if not qdir:
-        res = 3
-    else:
-        qdir_path = idxctx.config.get_valid_dir_option(
-            "QDIR", Path(qdir), idxctx.logger
-        )
-        if not qdir_path:
-            res = 3
-
-    if res != 0:
-        # Exit early if we encounter any errors.
-        return res
+class LinkSrcDestObj:
```

Comment:
This is just setting up some fixed instance data based on the options; it would make more sense abstracted into the initialization of an `Index` Class instance that would also encapsulate your `process` and `collect` functions. That would get the "orchestration" and linkage data between the two out of `main`. Essentially, `main` would just process the options, instantiate an object, and invoke it to do the work.